include/unifex/variant_sender.hpp

```diff
@@ -0,0 +1,124 @@
+namespace _variant_sender {
+
+template <typename... Ops>
+struct _op {
+  struct type;
+};
+template <typename... Ops>
+using operation = typename _op<Ops...>::type;
+
+template <typename... Ops>
+struct _op<Ops...>::type {
+  void start() & noexcept {
+    std::visit([](auto& op) noexcept(noexcept(unifex::start(op))) { unifex::start(op); }, variant_op);
+  }
+
+  std::variant<Ops...> variant_op;
+};
+
+
+template <_block::_enum First, _block::_enum... Rest>
+struct max_blocking_kind {
+  constexpr auto operator()() noexcept { return First; }
+};
+
+template <_block::_enum First, _block::_enum Second, _block::_enum... Rest>
+struct max_blocking_kind<First, Second, Rest...> {
+  constexpr auto operator()() noexcept {
+    if constexpr (First == Second) {
+      return max_blocking_kind<First, Rest...>{}();
+    } else if constexpr (
+        (First == blocking_kind::always &&
+        Second == blocking_kind::always_inline) ||
+        (Second == blocking_kind::always &&
+        First == blocking_kind::always_inline)) {
+      return max_blocking_kind<blocking_kind::always_inline, Rest...>{}();
+    } else {
+      return blocking_kind::maybe;
+    }
+  }
+};
+
+template <typename... Senders>
+struct _sender {
+  class type;
+};
+template <typename... Senders>
+using sender = typename _sender<Senders...>::type;
+
+template <typename... Senders>
+class _sender<Senders...>::type {
+  std::variant<Senders...> sender_variant;
+
+ public:
+
+  template <
+      template <typename...> class Variant,
+      template <typename...> class Tuple>
+  using value_types = typename concat_type_lists_unique_t<sender_value_types_t<Senders, type_list, Tuple>...>::template apply<Variant>;
+
+  template <template <typename...> class Variant>
+  using error_types = concat_type_lists_unique_t<sender_error_types_t<Senders, Variant>...>;
+
+  static constexpr bool sends_done = std::disjunction_v<std::bool_constant<sender_traits<Senders>::sends_done>...>;
+
+  template<typename ConcreteSender>
+  type(ConcreteSender&& concrete_sender)
+    noexcept(std::is_nothrow_constructible_v<std::variant<Senders...>, decltype(concrete_sender)>)
+    : sender_variant(std::forward<ConcreteSender>(concrete_sender)) {}
+
+  template(typename This, typename Receiver)
+    (requires same_as<remove_cvref_t<This>, type> AND std::conjunction_v<std::bool_constant<sender_to<member_t<This, Senders>, Receiver>>...>)
+  friend auto tag_invoke(tag_t<connect>, This&& that, Receiver&& r)
+    noexcept(std::conjunction_v<unifex::is_nothrow_connectable<Senders, Receiver>...>)
+  {
+    return operation<connect_result_t<Senders, Receiver>...>{
+        std::visit([r = static_cast<Receiver&&>(r)](auto&& sender) mutable noexcept(unifex::is_nothrow_connectable_v<decltype(sender), Receiver>) {
+            return std::variant<connect_result_t<Senders, Receiver>...>{unifex::connect(std::move(sender), static_cast<Receiver&&>(r))};
+        }, std::move(static_cast<This&&>(that).sender_variant))
```

Comment:
I think you're moving the sender when you should be forwarding it here.
```suggestion
            return std::variant<connect_result_t<Senders, Receiver>...>{unifex::connect(static_cast<decltype(sender)>(sender), static_cast<Receiver&&>(r))};
        }, static_cast<This&&>(that).sender_variant)
```